src/utils/entitySpec.js

```diff
@@ -139,4 +139,49 @@ export default class EntitySpec {
   toString() {
     return this.toPath()
   }
+
+  static getComponentUrl(coordinates) {
```

Comment:
I've moved these methods into a utils class to keep the original method more readable and to be able to re-use them somewhere else